vault/identity_store_aliases.go

```diff
@@ -181,10 +181,6 @@ func (i *IdentityStore) handleAliasUpdateCommon() framework.OperationFunc {
 			if entity == nil {
 				return nil, fmt.Errorf("existing alias is not associated with an entity")
 			}
-			if canonicalID == "" || entity.ID == canonicalID {
```

Comment:
Why is this taken out?

Reply:
This is to ensure that we always update the alias. This is useful when an existing alias name is updated with its casing altered.